lib/datasource/index.ts

```diff
@@ -149,17 +148,16 @@ async function mergeRegistries(
 
 function resolveRegistryUrls(
   datasource: Datasource,
-  extractedUrls: string[]
+  extractedUrls: string[] = []
 ): string[] {
   const { defaultRegistryUrls = [], appendRegistryUrls = [] } = datasource;
-  const customUrls = extractedUrls?.filter(Boolean);
-  let registryUrls: string[];
-  if (is.nonEmptyArray(customUrls)) {
-    registryUrls = [...extractedUrls, ...appendRegistryUrls];
-  } else {
-    registryUrls = [...defaultRegistryUrls, ...appendRegistryUrls];
-  }
-  return registryUrls.filter(Boolean);
+  return [
+    ...new Set(
+      [...defaultRegistryUrls, ...extractedUrls, ...appendRegistryUrls].filter(
```

Comment:
Actually I'm not sure which order is right here, but this one would allow safely remove default URLs from `manager/extract` code.

De-duplication seems to be good idea if we want to gradually refactor managers one by one.

Reply:
I don't see why there was a need to change, and this seems wrong